### PR TITLE
Allow orphan terms to be exported with `--allow_orphan_terms` flag

### PR DESCRIPTION
**Pull request by jacksonwp**
Fixes https://github.com/wp-cli/export-command/issues/108

This PR adds a flag `--allow_orphan_terms`, that when set allows the export to proceed when orphan terms are found.

When the flag is set, the orphaned terms have their parent set to 0 and the export proceeds.

When the flag is not set, an exception is thrown and the export fails.

Related https://github.com/wp-cli/wp-cli/issues/5832

### CODE REVIEW

File: src/WP_Export_Query.php [hunk 1]
```diff
@@ -368,10 +371,26 @@ private function check_for_orphaned_terms( $terms ) {
...
+				if ( $this->filters['allow_orphan_terms'] ) {
+					$orphans[ $has_parent->term_id ] = true;
+				} else {
+					$this->missing_parents = $has_parent;
+					throw new WP_Export_Term_Exception( "Term is missing a parent: {$has_parent->slug} ({$has_parent->term_taxonomy_id})" );
+				}
+			}
+		}
+
+		if ( count( $orphans ) > 0 ) {
```
Comment (danielbachhuber) [inline src/WP_Export_Query.php:383]: For the sake of clarity and avoiding accidental breakage, can we return `$terms` early if `! $this->filters['allow_orphan_terms']`?

**Review by danielbachhuber — CHANGES_REQUESTED** [review]
Looks pretty solid 👍 Just a few nits to polish up.

File: features/export.feature [hunk 2]
```diff
@@ -1148,3 +1148,26 @@ Feature: Export content.
...
+    Given a WP install
+    And I run `wp term create category orphan --parent=1`
+    And I run `wp db query "DELETE FROM wp_terms WHERE term_id = 1"`
+
+    When I run `wp export --allow_orphan_terms`
+    And save STDOUT 'Writing to file %s' as {EXPORT_FILE}
+    Then the {EXPORT_FILE} file should contain:
+      """
+      <wp:category_nicename>orphan</wp:category_nicename>
+      """
```
[the remark this answers is not in the document]
Comment (jacksonwp) [reply features/export.feature:1167]: @danielbachhuber - updated here: https://github.com/wp-cli/export-command/pull/109/commits/218ac7528957413b0d99b57f0de8072d27b41574

File: src/Export_Command.php [hunk 3]
```diff
@@ -65,6 +65,9 @@ class Export_Command extends WP_CLI_Command {
 	 * are categories, tags, nav_menu_items, custom_taxonomies_terms. Separate multiple
 	 * sections with a comma. Defaults to none.
 	 *
+	 * [--allow_orphan_terms]
+	 * : Don't throw an exception when orphaned terms are found.
```
[the remark this answers is not in the document]
Comment (jacksonwp) [reply src/Export_Command.php]: @danielbachhuber updated here: https://github.com/wp-cli/export-command/pull/109/commits/218ac7528957413b0d99b57f0de8072d27b41574

File: src/WP_Export_Query.php (hunk 1 above)
Comment (jacksonwp) [reply src/WP_Export_Query.php:388]: @danielbachhuber Updated here: https://github.com/wp-cli/export-command/pull/109/commits/218ac7528957413b0d99b57f0de8072d27b41574

**Review by danielbachhuber — APPROVED** [review]
Great work on this, @jacksonwp. I made a few additional tweaks you might want to look through.

**Pull request merged**